src/microbe_stage/MicrobeAI.cs

```diff
@@ -119,281 +74,109 @@ public void Think(float delta, Random random, MicrobeAICommonData data)
     {
         _ = delta;
 
-        // SetRandomTargetAndSpeed(random);
-
         // Clear the lists
-        predatoryMicrobes.Clear();
-        preyMicrobes.Clear();
         chunkList.Clear();
-
-        // only about 76.6 to 86.6 repeating seconds + or minus a few due to randomness at max focus
-        if (boredom >= (int)random.Next(SpeciesFocus / 2, 180 + (SpeciesFocus / 2)))
+        try
         {
-            if (prey != null && RollReverseCheck(SpeciesFocus, 600, random))
+            GetNearestPredatorItem(data.AllMicrobes);
+            targetChunk = GetNearestChunkItem(data.AllChunks, data.AllMicrobes);
+            var possiblePrey = GetNearestPreyItem(data.AllMicrobes);
+
+            if (microbe.IsBeingEngulfed)
             {
-                previousPrey = prey;
+                SetMoveSpeed(Constants.AI_BASE_MOVEMENT);
             }
-
-            if (targetChunk != null && RollReverseCheck(SpeciesFocus, 600, random))
+            else if (predator != null &&
+                DistanceFromMe(predator.Translation) < (1500.0 * SpeciesFear / Constants.MAX_SPECIES_FEAR))
             {
-                previousChunk = targetChunk;
+                PreyFlee(random);
             }
-
-            // Occasionally you need to reevaluate things
-            boredom = 0;
-            prey = null;
-            if (RollCheck(SpeciesActivity, 400, random))
+            else if (targetChunk != null &&
+                (targetChunk.Translation - microbe.Translation).LengthSquared()
+                <= (20000.0 * SpeciesFocus / Constants.MAX_SPECIES_FOCUS) + 1500.0)
+            {
+                PursueAndConsumeChunks(targetChunk, random);
+            }
+            else if (possiblePrey != null)
             {
-                lifeState = LifeState.PLANTLIKE_STATE;
+                EngagePrey(possiblePrey);
             }
             else
             {
-                lifeState = LifeState.NEUTRAL_STATE;
+                RunAndTumble(random);
             }
         }
-        else
+        catch
```

Comment:
Why do you have a plain catch block, this isn't very good programming practice? Why is this needed? You should detect the problem beforehand and just call RunAndTumble

Reply:
This was based on new cells not seeming to have everything loaded and just sitting around stationary, so I figured that trying to forcast what was going on with the microbe's initialization was going to be tightly coupling the AI to microbe setup (more that it already is), so I picked this admittedly lazy solution. I'm going to think on some kind of initial behavior of the AI to make the microbe not look like it's doing nothing.

Reply:
I picked a simple solution: https://github.com/adQuid/Thrive/blob/improve_microbe_ai/src/microbe_stage/MicrobeAI.cs#L54

This at least keeps microbes from spinning around a focus point already on themselves when they spawn in.